File: frontend/src/types/asset.ts
import type { AssetType } from './api'

export interface GrowthAsset {
  asset_id: string
  session_id: string
  quest_id: string
  event_id: string
  asset_type: AssetType
  asset_name: string
  asset_content: string | null
  created_at: string
}

export interface AssetListResponse {
  items: GrowthAsset[]
}

File: backend/src/models/asset.py
"""成长资产 Pydantic 模型 — 对齐 api-contracts.md。"""

from pydantic import BaseModel


class GrowthAssetDTO(BaseModel):
    asset_id: str
    session_id: str
    quest_id: str
    event_id: str
    asset_type: str
    asset_name: str
    asset_content: str | None = None
    created_at: str


class AssetListResponseData(BaseModel):
    items: list[GrowthAssetDTO]
